pkg/controller/datadogagentdeployment/clusteragent.go

```diff
@@ -68,7 +68,15 @@ func (r *ReconcileDatadogAgentDeployment) reconcileClusterAgent(logger logr.Logg
 		}
 
 		result, err := r.updateClusterAgentDeployment(logger, dad, clusterAgentDeployment, newStatus)
-		return result, err
+		if err != nil {
+			return result, err
+		}
+
+		if clusterAgentDeployment.Status.ReadyReplicas == 0 {
+			return reconcile.Result{Requeue: true}, fmt.Errorf("cluster agent deployment is not ready yet: 0 pods ready out of %d", clusterAgentDeployment.Status.Replicas)
```

Comment:
would it be better to use `RequeueAfter` instead ?

Reply:
I think we don't need to requeue at all since we are watching the ClusterAgent deployment resource events.

Reply:
makes sense